aikau/src/main/resources/alfresco/forms/controls/MultiSelect.js

```diff
@@ -0,0 +1,1230 @@
+            while (this.resultsDropdown.childNodes.length > this._results.length + 3) {
+               this.resultsDropdown.removeChild(this.resultsDropdown.lastChild);
+            }
+
+            // Update or add as necessary
+            array.forEach(this._results, function(nextResult, index) {
+
+               // Setup variables
+               var resultIsChosen = array.some(this._choices, function(nextChoice) {
+                     return nextChoice.value === nextResult.value;
+                  }),
+                  itemNode = this.resultsDropdown.childNodes[index + 3],
+                  clickListener,
+                  mouseoverListener;
+
+               // Construct the item if not already present
+               if (!itemNode) {
+                  itemNode = domConstruct.create("li", {
+                     "className": this.rootClass + "__result"
+                  }, this.resultsDropdown);
+               }
+
+               // Update the value if necessary
+               if (itemNode.getAttribute(this._valueHtmlAttribute) !== nextResult.value) {
+                  itemNode.setAttribute(this._valueHtmlAttribute, nextResult.value);
+               }
+
+               // Recreate the label
+               var newLabel = this._createHighlightedLabel(nextResult.label);
+               if (itemNode.innerHTML !== newLabel.outerHTML) {
+                  while (itemNode.hasChildNodes()) {
+                     itemNode.removeChild(itemNode.firstChild);
+                  }
+                  itemNode.appendChild(newLabel);
+               }
+
+               // Setup event listeners
+               clickListener = on(itemNode, "mousedown", lang.hitch(this, this._onResultMousedown));
+               mouseoverListener = on(itemNode, "mouseover", lang.hitch(this, this._onResultMouseover));
+               this._resultListeners.push(clickListener, mouseoverListener);
+               this.own(clickListener, mouseoverListener);
+
+               // Update the domNode in the result item
+               nextResult.domNode = itemNode;
+
+               // Mark item as chosen, if appropriate and remove focused indicator
+               domClass[resultIsChosen ? "add" : "remove"](itemNode, this.rootClass + "__result--already-chosen");
+               domClass.remove(itemNode, this.rootClass + "__result--focused");
+
+            }, this);
+         },
+
+         /**
+          * Update all of the items with latest info from the store
+          *
+          * @instance
+          * @protected
+          */
+         _updateItemsFromStore: function alfresco_forms_controls_MultiSelect___updateItemsFromStore() {
+
+            // Make sure we have some that need updating
+            if (!this._itemsToUpdateFromStore.length) {
+               return;
+            }
+
+            // Setup handlers
+            var successHandler = lang.hitch(this, function(responseItems) {
```

Comment:
Is there a possibility that someone (us or a 3rd party) might want to override the code in this closure? If so, we should probably move it out to a widget level function.